### PR TITLE
tests: Improved tests

### PR DESCRIPTION
**Pull request by Xartos**
> [!warning]
> **This is a public repository, ensure not to disclose:**
>
> - [ ] personal data beyond what is necessary for interacting with this pull request, nor
> - [ ] business confidential information, such as customer names.

### What kind of PR is this?

**Required**: Mark one of the following that is applicable:

- [ ] kind/feature       
- [x] kind/improvement   
- [ ] kind/deprecation   
- [ ] kind/documentation 
- [ ] kind/clean-up      
- [ ] kind/bug           
- [ ] kind/other         

*Optional*: Mark one or more of the following that are applicable:

> [!important]
> Breaking changes should be marked `kind/admin-change` or `kind/dev-change` depending on type
> Critical security fixes should be marked with `kind/security`

- [ ] kind/admin-change 
- [ ] kind/dev-change   
- [ ] kind/security     
- [ ] [kind/adr]()      

### What does this PR do / why do we need this PR?

I found that it was quite unclear what happened when the ingress tests failed while using proxy protocol, so I changed the test to tell the user that the test were skipped rather than failed.
I found an issue where the tests would error out if there were no admin users. So I fixed that and also added so that the test tested admin groups as well.

#### Checklist

- [ ] Proper commit message prefix on all commits
  
- Change checks:
  - [ ] The change is transparent
  - [ ] The change is disruptive
  - [ ] The change requires no migration steps
  - [ ] The change requires migration steps
  - [ ] The change upgrades CRDs
  - [ ] The change updates the config *and* the schema
- Metrics checks:
  - [ ] The metrics are still exposed and present in Grafana after the change
  - [ ] The metrics names didn't change (Grafana dashboards and Prometheus alerts are not affected)
  - [ ] The metrics names did change (Grafana dashboards and Prometheus alerts were fixed)
- Logs checks:
  - [ ] The logs do not show any errors after the change
- Pod Security Policy checks:
  - [ ] Any changed pod is covered by Pod Security Admission
  - [ ] Any changed pod is covered by Gatekeeper Pod Security Policies
  - [ ] The change does not cause any pods to be blocked by Pod Security Admission or Policies
- Network Policy checks:
  - [ ] Any changed pod is covered by Network Policies
  - [ ] The change does not cause any dropped packets in the `NetworkPolicy Dashboard`
- Audit checks:
  - [ ] The change does not cause any unnecessary Kubernetes audit events
  - [ ] The change requires changes to Kubernetes audit policy
- Falco checks:
  - [ ] The change does not cause any alerts to be generated by Falco
- Bug checks:
  - [ ] The bug fix is covered by regression tests

### CODE REVIEW

File: pipeline/test/services/workload-cluster/testHNC.sh [hunk 1]
```diff
@@ -43,7 +43,8 @@ function check_wc_hnc_creation_removal() {
     debug_msg=""
 
     mapfile -t user_namespaces < <(yq4 -e '.user.namespaces[]' "${config['config_file_wc']}")
-    mapfile -t user_admin_users < <(yq4 -e '.user.adminUsers[]' "${config['config_file_wc']}")
+    mapfile -t user_admin_users < <(yq4 -e '.user.adminUsers[] // ""' "${config['config_file_wc']}")
```
Comment (davidumea) [inline pipeline/test/services/workload-cluster/testHNC.sh:46]: Why was this changed? I get this output as a result
image

Comment (Xartos) [reply pipeline/test/services/workload-cluster/testHNC.sh:46]: Because if the list was empty it would fail for me. But this is not the behaviour I'm seeing, I guess it might be some yq or bash version difference :thinking:

**Comment by Xartos** [comment]
Anyone else having this issue :point_up:

**Comment by Xartos** [comment]
> Anyone else having this issue? ☝️

If more people have this issue, you are welcome to take over this PR and find a solution

**Comment by viktor-f** [comment]
This is now updated so that the list of users and/or groups can be empty.
Also skips checking hnc for namespaces with extra permissions since they are disabled in hnc.

File: pipeline/test/services/workload-cluster/testHNC.sh [hunk 2]
```diff
@@ -42,8 +42,10 @@ function check_wc_hnc_creation_removal() {
     no_error=true
     debug_msg=""
 
-    mapfile -t user_namespaces < <(yq4 -e '.user.namespaces[]' "${config['config_file_wc']}")
-    mapfile -t user_admin_users < <(yq4 -e '.user.adminUsers[]' "${config['config_file_wc']}")
+    # shellcheck disable=SC2016
+    mapfile -t user_namespaces < <(yq4 -e '(.user.constraints | keys | .[]) as $nonHNCNamespaces | .user.namespaces[] | select(. != $nonHNCNamespaces)' "${config['config_file_wc']}")
```
Comment (aarnq) [inline pipeline/test/services/workload-cluster/testHNC.sh:46]: Have you tested this with multiple namespaces under constraints?

Not sure if our version supports this but `.user.namespaces - (.user.constraints | keys) | .[]` _should_ give you what you need.

Comment (viktor-f) [reply pipeline/test/services/workload-cluster/testHNC.sh]: I tried your version with multiple namespaces under constraints and it worked well.
Thanks for the suggestion. It does feel like there is always a simpler way to write yq commands.

**Review by aarnq — APPROVED** [review]

**Review by Pavan-Gunda — APPROVED** [review]
